src/jit/codegenxarch.cpp

```diff
@@ -2747,61 +2745,69 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             emit->emitIns_R_L(INS_lea, EA_PTR_DSP_RELOC, genPendingCallLabel, treeNode->gtRegNum);
             break;
 
-        case GT_COPYOBJ:
-            genCodeForCpObj(treeNode->AsCpObj());
-            break;
+        case GT_STORE_OBJ:
+            if (treeNode->OperIsCopyBlkOp() && !treeNode->AsBlk()->gtBlkOpGcUnsafe)
+            {
+                assert(treeNode->AsObj()->gtGcPtrCount != 0);
+                genCodeForCpObj(treeNode->AsObj());
+                break;
+            }
+            __fallthrough;
 
-        case GT_COPYBLK:
+        case GT_STORE_DYN_BLK:
+        case GT_STORE_BLK:
         {
```

Comment:
This function is already huge. Couldn't we refactor this case out as genCodeForStoreBlk()?